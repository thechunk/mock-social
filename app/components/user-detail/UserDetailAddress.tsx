import * as React from 'react';
import {PureComponent} from "react";
import {TouchableWithoutFeedback, View, Text} from "react-native";
import styles from "./styles";
import g, {Color} from '../../styles/global';
import {NavigationInjectedProps, withNavigation} from "react-navigation";
import Icon from 'react-native-vector-icons/Ionicons';
import withNavigationParamsAsProps from "../../hoc/withNavigationParamsAsProps";

export class UserDetailAddress extends PureComponent<IUserDetailAddress & NavigationInjectedProps &
    IWithNavigationParamsAsProps> {
    constructor(props: IUserDetailAddress & NavigationInjectedProps & IWithNavigationParamsAsProps) {
        super(props);
        this.onAddressPress = this.onAddressPress.bind(this);
    }

    onAddressPress() {
        this.props.navigation.navigate('MapView', {hydrate: this.props.user});
    }

    render() {
        return (
            <View style={[styles.callOutSection, g.roundedBorder]}>
                <TouchableWithoutFeedback onPress={this.onAddressPress}>
                    <View style={styles.callOutAddress}>
                        <View style={g.flex1}>
                            <Text>
                                {`${this.props.user.address.suite} ${this.props.user.address.street}`}
                            </Text>
                            <Text>{this.props.user.address.city}</Text>
                            <Text>{this.props.user.address.zipcode}</Text>
                        </View>
                        <Icon
                            style={styles.callOutAccessory}
                            name="ios-arrow-dropright"
                            size={30}
                            color={Color.SecondaryAccent} />
                    </View>
                </TouchableWithoutFeedback>
            </View>
        )
    }
}

export default withNavigation(withNavigationParamsAsProps(UserDetailAddress));