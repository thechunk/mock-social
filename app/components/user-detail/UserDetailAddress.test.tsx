import * as React from 'react';
import {UserDetailAddress} from './UserDetailAddress';

import {shallow} from 'enzyme';

describe('UserDetailAddress', () => {
    const user = {address: {suite: '123', street: 'Fake St.', city: 'Springfield', zipcode: '12345'}};
    const navigation = {navigate: jest.fn()};

    beforeEach(() => {
        navigation.navigate.mockClear();
    });

    it('renders correctly', () => {
        expect(shallow(<UserDetailAddress user={user} navigation={navigation} />))
            .toMatchSnapshot();
    });

    it('navigates to MapView on address tap', () => {
        const wrapper = shallow(<UserDetailAddress user={user} navigation={navigation} />);
        wrapper.instance().onAddressPress();
        expect(navigation.navigate).toBeCalledWith('MapView', {hydrate: user});
    });
});
