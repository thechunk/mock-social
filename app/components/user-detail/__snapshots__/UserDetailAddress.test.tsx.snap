// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailAddress renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "marginBottom": 12,
        "marginLeft": 12,
        "marginRight": 12,
        "padding": 24,
      },
      Object {
        "borderBottomLeftRadius": 12,
        "borderBottomRightRadius": 12,
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
        "overflow": "hidden",
      },
    ]
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text>
          123 Fake St.
        </Text>
        <Text>
          Springfield
        </Text>
        <Text>
          12345
        </Text>
      </View>
      <[object Object]
        color="#c06171"
        name="ios-arrow-dropright"
        size={30}
        style={
          Object {
            "flex": 0,
          }
        }
      />
    </View>
  </TouchableWithoutFeedback>
</View>
`;
